tests/cpp/test_gpu_tensorcore.cpp

```diff
@@ -1671,175 +1671,215 @@ TEST_F(GPUTTensorCoreTest, FusionAmpereViewMatmulTN_CUDA) {
+    //
+    // [Mo,Ko,M,K]
+    // Swizzle tv0: 128 x 32 tile:
+    tv0cw->split(-2, 8);
+    tv0cw->split(-2, 2);
+    tv0cw->split(-1, 8);
+    //        -5   -4 -3 -2 -1
+    // [Mo,Ko,Mo16,M4,M2,Ko4,K8]
+    tv0cw->swizzle(Swizzle2DType::XOR, -4, -2);
+    tv0cw->merge(-4);
+    tv0cw->merge(-3);
+    //         -3   -2  -1
+    // [Mo,Ko,Mo16,warp,K8]
+    tv0cw->split(-3, 4);
+    tv0cw->split(-3, 2);
+    //             -4  -3   -2  -1
+    // [Mo,Ko, S4, wz2, wy2, warp,K8]
+    tv0cw->axis(-4)->parallelize(ParallelType::TIDz);
+    tv0cw->axis(-3)->parallelize(ParallelType::TIDy);
+    tv0cw->axis(-2)->parallelize(ParallelType::TIDx);
+    tv0cw->axis(-1)->parallelize(ParallelType::Vectorize);
+
+    tv0cw->setMemoryType(MemoryType::Shared);
+    // [Mo,Ko,i,wy,wx,v]
+
+    // [No,Ko,N,K]
+    // Swizzle tv0: 128 x 32 tile:
+    tv1cw->split(-2, 8);
+    tv1cw->split(-2, 2);
+    tv1cw->split(-1, 8);
+    //        -5   -4 -3 -2 -1
+    // [No,Ko,No16,N4,N2,Ko4,K8]
+    tv1cw->swizzle(Swizzle2DType::XOR, -4, -2);
+    tv1cw->merge(-4);
+    tv1cw->merge(-3);
+    //         -3   -2  -1
+    // [No,Ko,No16,warp,K8]
+    tv1cw->split(-3, 4);
+    tv1cw->split(-3, 2);
+    //             -4  -3   -2  -1
+    // [No,Ko, S4, wz2, wy2, warp,K8]
+    tv1cw->axis(-4)->parallelize(ParallelType::TIDz);
+    tv1cw->axis(-3)->parallelize(ParallelType::TIDy);
+    tv1cw->axis(-2)->parallelize(ParallelType::TIDx);
+    tv1cw->axis(-1)->parallelize(ParallelType::Vectorize);
+
+    tv1cw->setMemoryType(MemoryType::Shared);
+    // Schedule mma input
+    tv0cr->applyMmaSwizzle(MmaOperand::A);
+
+    // [... Mi, Ni, Ki]
+    tv0b->reorder({{-2, -3}, {-3, -2}});
+    tv0b->applyMmaSwizzle(MmaOperand::A);
+
+    tv1cr->applyMmaSwizzle(MmaOperand::B);
+    tv1b->applyMmaSwizzle(MmaOperand::B);
+
+    // Schedule mma output
+    tv2c->applyMmaSwizzle(MmaOperand::Accumulator);
+    tv2->applyMmaSwizzle(MmaOperand::Accumulator);
+
+    // Parallelize
+    //  0   1  2  3   4   5  6   7  8  9  10
+    // [Mo No Ko Kwo Mwo Nwo Mw Nw (Mi Ni Ki)]
+    tv2c->axis(4)->parallelize(ParallelType::TIDz);
+    tv2c->axis(5)->parallelize(ParallelType::TIDy);
+
+    // Parallelize
+    //  0  1  2   3   4   5  6  7
+    // [Mo No Mwo Nwo Mw Nw (Mi Ni)]
+    tv2->axis(0)->parallelize(ParallelType::BIDx);
+    tv2->axis(1)->parallelize(ParallelType::BIDy);
+    tv2->axis(2)->parallelize(ParallelType::TIDz);
+    tv2->axis(3)->parallelize(ParallelType::TIDy);
+
+    tv0cw->doubleBuffer();
+    tv1cw->doubleBuffer();
+    tv0cr->doubleBuffer();
+    tv1cr->doubleBuffer();
+
+    // Create input for either [M, K] @ [N, K] or [M, K] @ [K, N] (with strides)
+    auto options = at::TensorOptions().dtype(at::kHalf).device(at::kCUDA, 0);
+    auto t0 = at::randn({M, K}, options);
+    auto t1 = use_mkn_dim_order ? at::randn({K, N}, options)
+                                : at::randn({N, K}, options);
+    if (use_mkn_dim_order) {
+      t1 = t1.as_strided(t1.sizes(), {1, K});
```

Comment:
nitpick, this can be merged with line 1861. no need for another conditional.

Reply:
I was trying to do that, but ran out of ideas to do it cleanly.  I can add it when building torch options, but that would repeating the use of the options builder a couple of times. What would you suggest?

Reply:
I was merely suggesting either with
```suggestion
    auto t1 = use_mkn_dim_order ? at::randn({K, N}, options).as_strided({K, N}, {1, K})
                                : at::randn({N, K}, options);
```

alternatively, the other case is really just a permuted tensor?
```
t1 = at::randn({N, K}, options);
if (use_mkn_dim_order) {
  t1 = t1.permute({1, 0});
}
```